src/sources/syslog.rs

```diff
@@ -27,6 +29,7 @@ pub struct SyslogConfig {
     #[serde(default = "default_max_length")]
     pub max_length: usize,
     pub host_key: Option<String>,
+    pub emit_structured: bool,
```

Comment:
I'm actually not sure we need to expose this as an option. I think I'd rather always insert the data into the event and then let the sink encoder decide if it wants to output it or not.